mergify_engine/branch_updater.py

```diff
@@ -44,20 +45,23 @@ class AuthentificationFailure(Exception):
     pass
 
 
-GIT_MESSAGE_TO_EXCEPTION = {
-    b"Invalid username or password": AuthentificationFailure,
-    b"Repository not found": AuthentificationFailure,
-    b"The requested URL returned error: 403": AuthentificationFailure,
-    b"Patch failed at": BranchUpdateFailure,
-    b"remote contains work that you do": BranchUpdateNeedRetry,
-    b"remote end hung up unexpectedly": BranchUpdateNeedRetry,
-    b"cannot lock ref 'refs/heads/": BranchUpdateNeedRetry,
-    b"Could not resolve host": BranchUpdateNeedRetry,
-    b"Operation timed out": BranchUpdateNeedRetry,
-    b"No such device or address": BranchUpdateNeedRetry,
-    b"Protected branch update failed": BranchUpdateFailure,
-    b"Couldn't find remote ref": BranchUpdateFailure,
-}
+GIT_MESSAGE_TO_EXCEPTION = collections.OrderedDict(
```

Comment:
Why does this need to be ordered?

Reply:
SAML error also match 'The requested URL returned error, 403'. So first match first raise.